File: graphql-test/src/main/java/com/junyss/graphqltest/people/model/dto/request/PeopleRequestDto.java
package com.junyss.graphqltest.people.model.dto.request;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

import com.junyss.graphqltest.common.enumtype.BloodType;
import com.junyss.graphqltest.common.enumtype.Role;
import com.junyss.graphqltest.common.enumtype.Sex;

import lombok.Data;

@Data
public class PeopleRequestDto {
	@NotBlank private Long teamId;
	@NotBlank private String lastName;
	@NotBlank private String firstName;
	@NotBlank private Sex sex;
	@NotBlank private BloodType bloodType;
	@NotNull @Min(0) private Integer serveYears;
	@NotBlank private Role role;
	@NotBlank private String hometown;
}

File: type-script-nest-graph-ql-실습/src/app/people/model/dto/request/people-request.dto.ts
import { Field, InputType } from "@nestjs/graphql";
import { IsNotEmpty, IsNumber, IsString, Min } from "class-validator";
import { Sex } from "../../../../../common/enum/people.sex.enum";
import { BloodType } from "../../../../../common/enum/people.blood-type.enum";
import { Role } from "../../../../../common/enum/people.role.enum";
import { PeopleEntity } from "../../entities/people.entity";
import { TeamEntity } from "../../../../team/model/entities/team.entity";

@InputType()
export class PeopleRequestDto {
  @IsNotEmpty()
  @IsNumber()
  @Min(1)
  @Field(() => Number)
  teamId: number;

  @IsNotEmpty()
  @IsString()
  @Field(() => String)
  lastName: string;

  @IsNotEmpty()
  @IsString()
  @Field(() => String)
  firstName: string;

  @IsNotEmpty()
  @Field(() => Sex)
  sex: Sex;

  @IsNotEmpty()
  @Field(() => BloodType)
  bloodType: BloodType;

  @IsNotEmpty()
  @IsNumber()
  @Field(() => Number)
  serveYears: number;

  @Field(() => Role)
  role: Role;

  @IsNotEmpty()
  @IsString()
  @Field(() => String)
  hometown: string;

  toEntity(peopleRequestDto: PeopleRequestDto): PeopleEntity {
    const people = new PeopleEntity();
    people.team = new TeamEntity();
    people.team.teamId = peopleRequestDto.teamId;
    people.lastName = peopleRequestDto.lastName;
    people.firstName = peopleRequestDto.firstName;
    people.sex = peopleRequestDto.sex;
    people.bloodType = peopleRequestDto.bloodType;
    people.serveYears = peopleRequestDto.serveYears;
    people.role = peopleRequestDto.role;
    people.hometown = peopleRequestDto.hometown;

    return people;
  }
}
